osf/models/node.py

```diff
@@ -2430,6 +2433,29 @@ def _remove_from_associated_collections(self, auth=None, force=False):
                     force=True
                 )
 
+    def get_addon(self, name, is_deleted=False):
+        request, user_id = get_request_and_user_id()
+
+        if hasattr(request, 'user') and not isinstance(request.user, AnonymousUser) and waffle.flag_is_active(request, features.ENABLE_GV) and name not in ['osfstorage', 'wiki']:
```

Comment:
I don't think we need to guard this against the AnonymousUser. Unauthenticated users should still be able to see the addons for a public node.